include/exadg/vector_tools/arbitrary_solution_transfer.h

```diff
@@ -0,0 +1,167 @@
+                    dealii::VectorTools::EvaluationFlags::EvaluationFlags const flags =
+                      dealii::VectorTools::EvaluationFlags::avg) const
+  {
+    src.update_ghost_values();
+    const auto values =
+      dealii::VectorTools::point_values<n_components>(rpe, *dof_handler_src, src, flags);
+    src.zero_out_ghost_values();
+
+    fill_dof_vector_with_values<n_components>(dst, *dof_handler_dst, values);
+  }
+
+private:
+  std::vector<dealii::Point<dim>>
+  collect_mapped_support_points(dealii::DoFHandler<dim> const & dof_handler,
+                                dealii::Mapping<dim> const &    mapping) const
+  {
+    std::vector<dealii::Point<dim>> global_mapped_support_points;
+    std::vector<dealii::Point<dim>> mapped_support_points;
+
+    for(auto const & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      mapped_support_points.resize(cell->get_fe().get_generalized_support_points().size());
+
+      unsigned int idx = 0;
+      for(auto const & sp : cell->get_fe().get_generalized_support_points())
+        mapped_support_points[idx++] = mapping.transform_unit_to_real_cell(cell, sp);
+
+      global_mapped_support_points.insert(global_mapped_support_points.end(),
+                                          mapped_support_points.begin(),
+                                          mapped_support_points.end());
+    }
+
+    return global_mapped_support_points;
+  }
+
+  template<int n_components, typename VectorType, typename ValueType>
+  void
+  fill_dof_vector_with_values(VectorType &                    dst,
+                              dealii::DoFHandler<dim> const & dof_handler,
+                              std::vector<ValueType> const &  values) const
+  {
+    auto ptr = values.begin();
+    for(const auto & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      unsigned int n_dofs_per_component = cell->get_fe().get_generalized_support_points().size();
+
+      if constexpr(n_components == 1)
+      {
+        cell->set_dof_values(
+          dealii::Vector<typename VectorType::value_type>(ptr, ptr + n_dofs_per_component), dst);
+      }
+      else
+      {
+        cell->set_dof_values(multiple_components_to_dof_values<typename VectorType::value_type>(
+                               cell->get_fe(), n_dofs_per_component, ptr),
+                             dst);
+      }
+
+      ptr += n_dofs_per_component;
+    }
+  }
+
+  template<typename Number, typename InputIterator>
+  dealii::Vector<Number>
+  multiple_components_to_dof_values(dealii::FiniteElement<dim> const & fe,
+                                    unsigned int const                 n_dofs_per_comp,
+                                    InputIterator const                first) const
+  {
+    std::vector<dealii::Vector<double>> vec;
+    vec.reserve(fe.n_dofs_per_cell());
```

Comment:
why n_dofs_per_cell and not n_dofs_per_component?

Reply:
It is possible that there are multiple components at the same support point. Using generalized support points gives us a unique representation of the used support points. However, we might evaluate multiple components at the same support point and thus have to write each component to the dof vector, i.e. n_dofs_per_cell.

Reply:
Not sure if I understood it correctly. My question is why the loop below is over `n_dofs_per_comp` while you reserve for `n_dofs_per_cell`?